File: src/main/java/de/fluxparticle/animation/binding/BindingFunction.java
package de.fluxparticle.animation.binding;

import javafx.beans.binding.ObjectBinding;
import javafx.beans.value.ObservableValue;

import java.util.function.Function;

public class BindingFunction<T, R> extends ObjectBinding<R> {

    private final ObservableValue<T> value;

    private final Function<T, R> func;

    public BindingFunction(ObservableValue<T> value, Function<T, R> func) {
        this.value = value;
        this.func = func;

        bind(value);
    }

    @Override
    protected R computeValue() {
        return func.apply(value.getValue());
    }

}

File: src/main/kotlin/de/fluxparticle/animation/binding/BindingFunction.kt
package de.fluxparticle.animation.binding

import javafx.beans.binding.ObjectBinding
import javafx.beans.value.ObservableValue

class BindingFunction<T, R>(private val value: ObservableValue<T>, private val func: (T) -> R) : ObjectBinding<R>() {

    init {

        bind(value)
    }

    override fun computeValue(): R {
        return func(value.value)
    }

}
